From 480c4e4a1b2f10ef3d1aba101241b4d22594e0fe Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Tue, 16 Jul 2013 12:47:30 -0700
Subject: [PATCH 5/5] Add pmecc header to bootstrap binary

The bootstrap bin file needs a 52-word header prepended to it when in flash.
In order to support writing using nandwrite and uboot without added
complexity, we'll prepend this header at build time. This also requires
that sam-ba writes flash using the same way we write uboot and kernel; in
other words, we won't use the "Send bootfile" script anymore.

This change uses a static binary file.  It's configurable per board if needed,
and only the wb45n board (at91sam9x5ek) uses it, other configurations would
be unaffected.
---
 Makefile                    |    9 ++++++++-
 board/at91sam9x5ek/board.mk |    4 ++++
 board/pmecc_header.bin      |    1 +
 3 files changed, 13 insertions(+), 1 deletion(-)
 create mode 100644 board/pmecc_header.bin

diff --git a/Makefile b/Makefile
index 3f82f14..46756d0 100644
--- a/Makefile
+++ b/Makefile
@@ -221,7 +221,14 @@ $(AT91BOOTSTRAP): $(OBJS)
 	@echo "  LD        "$(BOOT_NAME).elf
 	@$(LD) $(LDFLAGS) -n -o $(BINDIR)/$(BOOT_NAME).elf $(OBJS)
 #	@$(OBJCOPY) --strip-debug --strip-unneeded $(BINDIR)/$(BOOT_NAME).elf -O binary $(BINDIR)/$(BOOT_NAME).bin
-	@$(OBJCOPY) --strip-all $(BINDIR)/$(BOOT_NAME).elf -O binary $@
+	@$(OBJCOPY) --strip-all $(BINDIR)/$(BOOT_NAME).elf -O binary $@_nohead.bin
+ifneq ($(PMECC_HEADER),)
+	@echo "Adding PMECC header"
+	@cat $(PMECC_HEADER) > $@
+	@cat $@_nohead.bin >> $@
+else
+	@mv $@_nohead.bin  $@
+endif
 
 %.o : %.c .config
 	@echo "  CC        "$<
diff --git a/board/at91sam9x5ek/board.mk b/board/at91sam9x5ek/board.mk
index f39ea81..83985cf 100644
--- a/board/at91sam9x5ek/board.mk
+++ b/board/at91sam9x5ek/board.mk
@@ -5,3 +5,7 @@ CPPFLAGS += \
 ASFLAGS += \
 	-DCONFIG_AT91SAM9X5EK \
 	-mcpu=arm926ej-s
+
+ifeq ($(CPU_HAS_PMECC),y)
+	PMECC_HEADER := "board/pmecc_header.bin"
+endif
diff --git a/board/pmecc_header.bin b/board/pmecc_header.bin
new file mode 100644
index 0000000..f8d6073
--- /dev/null
+++ b/board/pmecc_header.bin
@@ -0,0 +1 @@
+$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À$À
\ No newline at end of file
-- 
1.7.9.5

